src/DnsClient/Protocol/CertificateType.cs

```diff
@@ -0,0 +1,47 @@
+﻿namespace DnsClient.Protocol;
+
+/// <summary>
+/// Certificate type values
+/// </summary>
+/// <remarks>
+///  <seealso href="https://tools.ietf.org/html/rfc4398#section-2.1">RFC 4398 section 2.1</seealso>
+/// </remarks>
+public enum CertificateType
+{
+    /// <summary>
+    /// Reserved certificate type.
+    /// </summary>
+    Reserved = 0,
+    /// <summary>
+    /// X509 certificate
+    /// </summary>
+    X509 = 1,
```

Comment:
The enum names should match the RFC. 
The RFC also clearly states that the string representation can be either the integer or mnemonic defined in section 2.1:

```
    Value  Mnemonic  Certificate Type
         -----  --------  ----------------
             0            Reserved
             1  PKIX      X.509 as per PKIX
             2  SPKI      SPKI certificate
             3  PGP       OpenPGP packet
             4  IPKIX     The URL of an X.509 data object
             5  ISPKI     The URL of an SPKI certificate
             6  IPGP      The fingerprint and URL of an OpenPGP packet
             7  ACPKIX    Attribute Certificate
             8  IACPKIX   The URL of an Attribute Certificate
         9-252            Available for IANA assignment
           253  URI       URI private
           254  OID       OID private
           255            Reserved
     256-65279            Available for IANA assignment
   65280-65534            Experimental
         65535            Reserved
```

I'd change the following:

* `X509` should be `PKIX`, rest seems fine I think
* I'd also add `URI` and `OID`
* anything unassigned or reserved can be left empty
* the ToString method should output the Mnemonic if available, not the integer:  `string.Format("{0} {1} {2} {3}", (int)CertType` <- just remove the cast to int
   * same for Algorithm

Reply:
Good catch.  And there in section [2.2 of RFC 4398](https://datatracker.ietf.org/doc/html/rfc4398#section-2.2) is what I needed to re-read concerning text representation.

Reply:
Looks good now. Thanks!